apis/Google.Cloud.BigQuery.V2/Google.Cloud.BigQuery.V2.IntegrationTests/InsertTest.cs

```diff
@@ -491,5 +492,42 @@ public void CreateLoadJob_Parquet()
             var rows = table.ListRows().ToList();
             Assert.Equal(50, rows.Count);
         }
+
+        [Fact]
+        public void CreateLoadJobWithSession()
```

Comment:
I'm confused by this test. Why are you creating two load jobs? Is it so the first one creates the session? Is there no other way to create a session?

Reply:
yes the first one creates the session.

Reply:
It's still unclear to me why we're loading the same data twice though. What is this test aiming to demonstrate? Do we expect customers to creates two load jobs for the same table *with the same data*? If this is actually meant to be about "load two different chunks of data" (as a use case) then that's what the test should do IMO.

Reply:
Hey @jskeet , Basically the fist Job demonstrates the session creation part where we set CreateSession = True, the data loading part is just a sample. In the 2nd load job we are reusing the session created by the previous job by setting SessionId = Session id got from the previous step. 

The aim of the test is to test the session creation and session usage both should work as expected.

Reply:
But is this what a customer would actually do? I can see a few possible scenarios:

- Create a session explicitly, load data, and then just query from the temp table (specifying the session ID)
- Create a session explicitly, load data part 1, then load data part 2 [...] specifying the session ID, then query

I think it would be useful to have different tests for those cases - and for the second test, it should load different pieces of data (as that's what a customer would do).